### PR TITLE
Update to Sprout invoices integration fixes #112

### LINKED ISSUE #112: Sprout Invoices payment deposit doesn't work
Having a deposit on the form doesn't submit the invoice. 
=========================================================
Name           |   Deposit for form submission. 
Reporter       |   Elana D.
Submit Date |   6/25/2021
Summary      |   Deposit for form submission. 
Reproduction steps |  Set up integration with Sprout Invoices. Put a deposit on the form for the price and submit the form. 

image

An invoice will create but none of the line items will come over. 
image

### PR DESCRIPTION
**Pull request by avonville**
Updated Sprout invoices integration to allow single product, multi product, checkbox. and radio to convert to line items.

Added the payment status to be passed to created invoices. 

Elana has tested this, just need a code review.

### CODE REVIEW

File: includes/integrations/sprout-invoices/class-integration-sprout-invoices.php [hunk 1]
```diff
@@ -83,6 +83,15 @@ public function create_doc($entry_id, $form_id, $page_id)
...
             return;
         }
+
+        // $form_payment_data = weforms_get_form_payments( $form_id );
+
+        // error_log("Form Payment Data" . " = " . print_r($form_payment_data,1));
+
+        $payment_data = weforms_get_entry_payment( $entry_id );
+
+        error_log( $payment_data );
```
Comment (jamesros161) [inline includes/integrations/sprout-invoices/class-integration-sprout-invoices.php:93]: is this error_log supposed to be here?

**Review by jamesros161 — CHANGES_REQUESTED** [review]

File: includes/functions.php [hunk 2]
```diff
@@ -164,6 +164,27 @@ function weforms_get_form_payments( $form_id, $args = [] ) {
...
+ */
+function weforms_get_entry_payment( $entry_id ) {
+    global $wpdb;
+
+    $query = 'SELECT transaction_id FROM ' . $wpdb->prefix . 'weforms_payments' .
+        ' WHERE entry_id = ' . $entry_id;
+        error_log(print_r($query,1));
+
+    $payment = $wpdb->get_row( $query, $entry_id );
+    error_log(print_r($payment,1));
```
[the remark this answers is not in the document]
Comment (avonville) [reply includes/functions.php:182]: should have been removed. Will remove

File: includes/integrations/sprout-invoices/class-integration-sprout-invoices.php (hunk 1 above)
Comment (avonville) [reply includes/integrations/sprout-invoices/class-integration-sprout-invoices.php:93]: should have been removed. Will remove

Comment (avonville) [reply includes/integrations/sprout-invoices/class-integration-sprout-invoices.php:93]: weforms_get_entry_payment isnt going to be used at this time as well

**Review by jamesros161 — APPROVED** [review]

**Pull request merged**